modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Comments/CommentManager.cs

```diff
@@ -0,0 +1,42 @@
+﻿using JetBrains.Annotations;
+using System;
+using System.Threading.Tasks;
+using Volo.Abp;
+using Volo.Abp.Domain.Services;
+using Volo.CmsKit.Users;
+
+namespace Volo.CmsKit.Comments
+{
+    public class CommentManager : DomainService
+    {
+        protected ICommentEntityTypeDefinitionStore DefinitionStore { get; }
+
+        public CommentManager(ICommentEntityTypeDefinitionStore definitionStore)
+        {
+            DefinitionStore = definitionStore;
+        }
+
+        public virtual async Task<Comment> CreateAsync([NotNull] CmsUser creator,
+                                                       [NotNull] string entityType,
+                                                       [NotNull] string entityId,
+                                                       [NotNull] string text,
+                                                       [CanBeNull] Guid? repliedCommentId = null)
+        {
+            Check.NotNull(creator, nameof(creator));
+
```

Comment:
We should add null checks for entityType, entityId, and text as well.

Reply:
Those are checked in constructor of Comment entity. That's why I ignored from here. Should we ad here too? @ilkayilknur

Reply:
Yes, we should add it. The implementation of the might change in the future. We also check the entityType definition below. What happens if the entityType is null? We should add checks.

Reply:
Ok, it's done with 96e84af